addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,974 @@
+                configureListener((String) getConfig().get(LISTENER_PORT));
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+
+        unregisterTransceiverStatusListener(this);
+
+        disconnect(socketChannel);
+
+        try {
+            if (selector != null) {
+                selector.close();
+            }
+        } catch (IOException e) {
+            logger.error("An exception occurred while closing the selector: '{}'", e.getMessage());
+        }
+
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    private void establishConnection() {
+        lock.lock();
+        try {
+            if (getConfig().get(IP_ADDRESS) != null && getConfig().get(PORT_NUMBER) != null) {
+                socketChannel = connect((String) getConfig().get(IP_ADDRESS),
+                        ((BigDecimal) getConfig().get(PORT_NUMBER)).intValue());
+                try {
+                    Thread.sleep(((BigDecimal) getConfig().get(RESPONSE_TIME_OUT)).intValue());
+                } catch (NumberFormatException | InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    logger.debug("An exception occurred while putting a thread to sleep: '{}'", e.getMessage());
+                }
+                onConnectable(socketChannel);
+            }
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private synchronized void onUpdate() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0,
+                    ((BigDecimal) getConfig().get(REFRESH_INTERVAL)).intValue(), TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public void onConnectionLost() {
+        logger.debug("Updating thing status to OFFLINE.");
+        updateStatus(ThingStatus.OFFLINE);
+        for (TransceiverStatusListener listener : transceiverStatusListeners) {
+            listener.onBridgeDisconnected(this);
+        }
+        establishConnection();
+    }
+
+    public void onConnectionResumed() {
+        logger.debug("Updating thing status to ONLINE.");
+        configureTransceiver(socketChannel);
+        updateStatus(ThingStatus.ONLINE);
+        for (TransceiverStatusListener listener : transceiverStatusListeners) {
+            listener.onBridgeConnected(this);
+        }
+    }
+
+    private void configureListener(String listenerPort) {
+        try {
+            listenerChannel = ServerSocketChannel.open();
+            listenerChannel.socket().bind(new InetSocketAddress(Integer.parseInt(listenerPort)));
+            listenerChannel.configureBlocking(false);
+
+            logger.info("Listening for incoming connections on {}", listenerChannel.getLocalAddress());
+
+            synchronized (selector) {
```

Comment:
Maybe it's better not combine a lock in some other method with this synchronised, I also see other actions in the sector which are not synchronised is that all expected